cilium/cmd/version.go

```diff
@@ -16,20 +16,68 @@ package cmd
 
 import (
 	"fmt"
+	"os"
 
 	"github.com/cilium/cilium/pkg/version"
 
 	"github.com/spf13/cobra"
 )
 
+var printJSONVersion bool
+
 var versionCmd = &cobra.Command{
 	Use:   "version",
 	Short: "Print version information",
 	Run: func(cmd *cobra.Command, args []string) {
-		fmt.Printf("Cilium %s\n", version.Version)
+		getVersion(cmd, args)
 	},
 }
 
 func init() {
 	rootCmd.AddCommand(versionCmd)
+	AddMultipleOutput(versionCmd)
+}
+
+func getVersion(cmd *cobra.Command, args []string) {
+	// -o argument is set
+	if len(dumpOutput) > 0 {
+		data := struct {
+			Client version.CiliumVersion
+			Daemon version.CiliumVersion
+		}{
+			getClientVersionAsStruct(),
+			getDaemonVersionAsStruct(),
+		}
+		if err := OutputPrinter(data); err != nil {
+			os.Exit(1)
+		}
+		return
+	}
+	// default output
+	fmt.Printf("Client: %s\n", getClientVersionAsString())
+	fmt.Printf("Daemon: %s\n", getDaemonVersionAsString())
+}
+
+func getClientVersionAsString() string {
+	return version.Version
+}
+
+func getDaemonVersionAsString() string {
+	resp, err := client.Daemon.GetDebuginfo(nil)
+	if err != nil {
+		return "Not responding"
```

Comment:
Make this a constant (since it's also used in getDaemonVersionAsStruct() below) or return `nil` as your magic value.